diffusion/models/stable_diffusion.py

```diff
@@ -469,22 +462,16 @@ def _prepare_text_embeddings(self, prompt, tokenized_prompts, tokenized_pad_mask
                 raise NotImplementedError('SDXL not yet supported with precomputed embeddings')
 
         # duplicate text embeddings for each generation per prompt
-        bs_embed, seq_len, _ = prompt_embeds.shape
-        prompt_embeds = prompt_embeds.repeat(1, num_images_per_prompt, 1)  # type: ignore
-        prompt_embeds = prompt_embeds.view(bs_embed * num_images_per_prompt, seq_len, -1)
-
-        if self.mask_pad_tokens and tokenized_pad_mask is not None:
-            if len(tokenized_pad_mask.shape) == 3:
-                attention_mask = tokenized_pad_mask[:, 0]
-                for i in range(1, tokenized_pad_mask.shape[1]):
-                    attention_mask |= tokenized_pad_mask[:, i]
-                tokenized_pad_mask = attention_mask
-            tokenized_pad_mask = tokenized_pad_mask.repeat(1, num_images_per_prompt, 1)
-            tokenized_pad_mask = tokenized_pad_mask.view(bs_embed * num_images_per_prompt, seq_len)  # [B, 77]
+        prompt_embeds = _duplicate_tensor(prompt_embeds, num_images_per_prompt)
+
+        if self.mask_pad_tokens:
+            tokenized_pad_mask = _create_unet_attention_mask(tokenized_pad_mask)
+            tokenized_pad_mask = _duplicate_tensor(tokenized_pad_mask, num_images_per_prompt)
```

Comment:
I just realized that my code is not be right 😬 When duplicating with num_images_per_prompt, the duplicated prompt embeds will be consecutive, but this is not true for the tokenized_pad_mask due to the different dimensions. 

Two fixes come to mind:
1. quickest fix would be to `tokenized_pad_mask.unsqueeze(-1)`
2. The nasty looking but feels more principled fix in `_duplicate_tensor`: `tensor = tensor.repeat(1, num_images_per_prompt, *[1,]*len(tensor.shape[2:]))` and `tensor.view(batch_size * num_images_per_prompt, seq_len, *[-1,]*len(tensor.shape[2:]))`
Using the second means you can avoid the `.squeeze(-1)`

Results:
```
>>> t2 = torch.randn(2, 3, 4)
>>> t2
tensor([[[-1.4956,  0.5156,  1.3679, -0.6651],
         [ 0.5277,  0.8577,  3.6882,  1.7380],
         [ 0.7071, -1.2043,  0.6128, -0.3668]],

        [[ 0.3232,  0.3563,  0.0479,  0.2255],
         [-0.5160, -0.2547,  0.8563,  1.2211],
         [-0.3290,  0.5519,  0.0721, -0.2651]]])
>>> t2.repeat(1, 2, 1).view(2*2, 3, -1)
tensor([[[-1.4956,  0.5156,  1.3679, -0.6651],
         [ 0.5277,  0.8577,  3.6882,  1.7380],
         [ 0.7071, -1.2043,  0.6128, -0.3668]],

        [[-1.4956,  0.5156,  1.3679, -0.6651],
         [ 0.5277,  0.8577,  3.6882,  1.7380],
         [ 0.7071, -1.2043,  0.6128, -0.3668]],

        [[ 0.3232,  0.3563,  0.0479,  0.2255],
         [-0.5160, -0.2547,  0.8563,  1.2211],
         [-0.3290,  0.5519,  0.0721, -0.2651]],

        [[ 0.3232,  0.3563,  0.0479,  0.2255],
         [-0.5160, -0.2547,  0.8563,  1.2211],
         [-0.3290,  0.5519,  0.0721, -0.2651]]])
>>> t3 = torch.randn(2, 3)
>>> t3
tensor([[-0.5588, -0.9764, -1.1396],
        [ 1.1794,  0.9496, -0.2843]])
>>> t3.repeat(1, 2, 1).view(4, 3, -1).squeeze(-1)
tensor([[-0.5588, -0.9764, -1.1396],
        [ 1.1794,  0.9496, -0.2843],
        [-0.5588, -0.9764, -1.1396],
        [ 1.1794,  0.9496, -0.2843]])
```

Reply:
flipped a count and went with #2 =)